lib/dockerdaemon/cli.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2016-2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dockerdaemon
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/uber/kraken/utils/log"
+
+	"golang.org/x/net/context/ctxhttp"
+)
+
+const maxUnixSocketPathSize = len(syscall.RawSockaddrUnix{}.Path)
+const defaultTimeout = 32 * time.Second
+
+// DockerClient is a docker daemon client.
+type DockerClient interface {
+	ImagePull(ctx context.Context, registry, repo, tag string) error
+}
+
+type dockerClient struct {
+	version string // docker version
+	host    string // host that client connects to
+	scheme  string // http/https
+
+	addr     string       // client address
+	protocol string       // unix
+	basePath string       // base part of the url
+	client   *http.Client // opens http.transport
```

Comment:
Can we get rid of these comments and instead name the fields more aptly? E.g. `clientAddr` instead of `addr`.